### DISCUSSION TITLE
High Availability

### DISCUSSION THREAD

**Discussion by mike-stewart** [open-webui/open-webui#4403] [Ideas]
Are there any best practices for running Open WebUI with high availability (HA), running multiple replicas of the web server? This is important for ensuring service uptime.

I couldn't find anything in the docs, but could surmise that you can get part of the way there by using an external database with a DATABASE_URL. However, there are other things stored in the DATA_DIR including uploads, that seem to cause issues when running multiple replicas that each have their own data volume. To fully support an HA setup, this would require that the Open WebUI app is fully stateless with all persistence handled externally.

Is there any roadmap to support this, or any workarounds that others have had success with?

**Comment by tjbck** [comment]
Related: #4622 #4550

**Comment by mike-stewart** [comment]
@tjbck What about using object storage (e.g. S3) for file uploads? That would help remove the limitation that an upload on one replica is not accessible from another.

**Reply by tjbck** [reply]
Would make a great PR! Having it as an option from the admin settings would be amazing.

**Comment by TedLin1993** [comment]
Yeah... we cannot use Function because it does not store in the database, which prevents all additional features of open-webui from being available.
